akka-actor-tests/src/test/java/akka/japi/pf/ReceiveBuilderTest.java

```diff
@@ -209,10 +209,15 @@ public void shouldMatchByExternalPredicate(){
   @Test
   public void shouldMatchEquals() {
     Msg2 msg2 = new Msg2("foo");
+    final FI.TypedPredicate<Integer> greaterThanOrEqual18 = anyInt -> anyInt >= 18;
+    final FI.TypedPredicate<Integer> greaterThanOrEqual16 = anyInt -> anyInt >= 16;
     Receive rcv = ReceiveBuilder.create()
         .matchEquals(msg2, m -> result("match msg2"))
         .matchEquals("foo", m -> result("match foo"))
+        .matchEquals("bar", () -> result("match bar"))
         .matchEquals(17, m -> result("match 17"))
+        .matchEquals(20,greaterThanOrEqual18,m -> result("match 20"))
+        .matchEquals(16,greaterThanOrEqual16,() -> result("match 16"))
```

Comment:
Add some tests which are missing before .

Reply:
one thing that could be interesting to look into is how understandable compilation errors become when we add a lot of overloads

I'm not fully convinced that this convenience is worth adding. You can use `__` as variable name if you don't care about it

Reply:
hm,then both are two keypress `__` and `()`

Reply:
But from a Java user perspective, If I wrote a `__` in my code, my workmate will say `What!?`

Reply:
yeah, learning something new is terrible

(joking aside, I completely understand)

Reply:
Yes, My life will be much easier if my work currently is writing Scala, but for maintenance reason, we are using only Java now. But thanks god, we could use Java 8, but still much developer here don't want to get out the Java 6 pool.